ZipBuilder/Sources/ZipBuilder/FrameworkBuilder.swift

```diff
@@ -406,32 +419,68 @@ struct FrameworkBuilder {
       fatalError("Could not create framework directory while building framework \(framework). " +
         "\(error)")
     }
-    // Verify Firebase frameworks include an explicit umbrella header for Firebase.h.
-    let headersDir = podsDir.appendingPathComponents(["Headers", "Public", framework])
-    if framework.hasPrefix("Firebase"),
-      framework != "FirebaseCoreDiagnostics",
-      framework != "FirebaseUI" {
-      let frameworkHeader = headersDir.appendingPathComponent("\(framework).h")
-      guard fileManager.fileExists(atPath: frameworkHeader.path) else {
-        fatalError("Missing explicit umbrella header for \(framework).")
+
+    // Find the location of the public headers.
+    let anyArch = LaunchArgs.shared.archs[0] // any arch is ok, but need to make sure we're building it
+    let archivePath = thinArchives[anyArch]!
+    let headersDir = archivePath.deletingLastPathComponent().appendingPathComponent("Headers")
+
+    // Find CocoaPods generated umbrella header.
+    var umbrellaHeader = ""
+    if framework == "gRPC-Core" {
+      // TODO: Proper handling of podspec-specified module.modulemap files with customized umbrella
+      // headers. This is good enough for Firebase since it doesn't need these modules.
+      umbrellaHeader = "\(framework)-umbrella.h"
+    } else {
+      var umbrellaHeaderURL: URL
+      do {
+        let files = try fileManager.contentsOfDirectory(at: headersDir,
+                                                        includingPropertiesForKeys: nil).compactMap { $0.absoluteString }
+        let umbrellas = files.filter { $0.contains("umbrella.h") }
+        if umbrellas.count != 1 {
+          fatalError("Did not find exactly one umbrella header in \(headersDir).")
+        }
+        umbrellaHeaderURL = URL(string: umbrellas[0])!
```

Comment:
Nit: this could be cleaned up a little bit by using `guard` and `guard let` (I typically try to use `guard` whenever some sort of validation is going on. Also using `first` since it's optional.

```
guard let firstUmbrella = umbrellas.first, 
    let foundHeader = URL(string: firstUmbrella) else { /* error */ }

umbrellaHeaderURL = foundHeader
```

You may even be able to combine them if `URL(string:)` takes an optional String but I don't think it does.

Reply:
Done. Thanks!